### PR TITLE
Fix race condition in testCancelAllMutations

### PR DESCRIPTION
**Pull request by palpatim**
Fixed race condition waiting for operation queue to be emptied

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWSAppSyncUnitTests/MutationQueueTests.swift [hunk 1]
```diff
@@ -641,34 +641,57 @@ class MutationQueueTests: XCTestCase {
...
+                queuedMutationShouldNotBeInvoked.fulfill()
             }
         }
 
         try appSyncClient.clearCaches(options: ClearCacheOptions(clearMutations: true))
 
-        wait(for: [mutationPerformed], timeout: 6.0)
+        wait(for: mutationExpectations, timeout: 3.0)
 
         XCTAssertEqual(appSyncClient.queuedMutationCount, 0)
```
Comment (lawmicha) [inline AWSAppSyncUnitTests/MutationQueueTests.swift]: would be nice to see use of assert overload with message like "The mutation queue should be emptied"

Comment (lawmicha) [inline AWSAppSyncUnitTests/MutationQueueTests.swift]: would be nice to see use of assert overload with message like "The mutation queue should be emptied"

File: AWSAppSyncUnitTests/MutationQueueTests.swift [hunk 2]
```diff
@@ -641,34 +641,57 @@ class MutationQueueTests: XCTestCase {
...
 
         mockHTTPTransport.operationResponseDelay = 0
 
+        // Test 2
         let furtherMutationPerformed = expectation(description: "Further post added")
         appSyncClient.perform(mutation: addPost) { result, error in
             XCTAssertEqual(result?.data?.createPostWithoutFileUsingParameters?.id, "TestPostID")
```
[the remark this answers is not in the document]
Comment (palpatim) [reply AWSAppSyncUnitTests/MutationQueueTests.swift]: Good callout. I'll take that on as a future enhancement.

File: AWSAppSyncUnitTests/MutationQueueTests.swift (hunk 1 above)
Comment (palpatim) [reply AWSAppSyncUnitTests/MutationQueueTests.swift]: See comment above--I think this is a dupe

Comment (lawmicha) [reply AWSAppSyncUnitTests/MutationQueueTests.swift]: yes for the `XCTAssertEqual`, okay makes sense! it was just initial intuition on my part.

just thinking out loud here, i wonder if this improves the test durability with an assert on the first mutation being executed.
```
// makes sure the first mutation was added is completed
wait(for: firstMutationExpectation, timeout: 1.0)

// check that the total mutations minus one is in the queue
XCTAssertEqual(appSyncClient.queuedMutationCount, expectedQueuedMutationCount)

// clear the cache
try appSyncClient.clearCaches(...)

// make sure the inverted expectations are not fulfilled after mutation queue is cleared
wait(for: mutationExpectations, timeout: 3.0)

// make sure there are no queued mutations
XCTAssertEqual(appSyncClient.queuedMutationCount, 0)
```

The reasoning is, seems unlikely, but there is a race condition on clearing the cache and asserting/fulfilling the first mutation.

**Pull request merged**